clients/src/main/java/org/apache/kafka/common/security/plain/PlainSaslServer.java

```diff
@@ -160,7 +160,7 @@ public SaslServer createSaslServer(String mechanism, String protocol, String ser
 
         @Override
         public String[] getMechanismNames(Map<String, ?> props) {
-            String noPlainText = (String) props.get(Sasl.POLICY_NOPLAINTEXT);
+            String noPlainText = (props != null) ? (String) props.get(Sasl.POLICY_NOPLAINTEXT) : "false";
```

Comment:
I think I'd just do a check at the top and say `if (props == null) return new String[0];` or something like that.